src/OpenTelemetry.Exporter.Zipkin/ZipkinExporter.cs

```diff
@@ -149,14 +142,20 @@ internal ZipkinEndpoint EnsureLocalEndpoint(Activity activity)
                 serviceName = this.options.ServiceName;
             }
 
-            return this.LocalEndpoint = new ZipkinEndpoint(
+            this.LocalEndpoint = new ZipkinEndpoint(
                 serviceName,
                 ipv4,
                 ipv6,
                 port: null,
                 tags);
         }
 
+        /// <inheritdoc/>
+        protected override void OnTracerProviderSet(TracerProvider tracerProvider)
+        {
+            this.SetResource(tracerProvider.GetResource());
```

Comment:
Do we expect `OnTracerProviderSet` to be used for scenarios other than `SetResource`?

Reply:
That was the idea ya. [Option 2](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1447) was specific to just resources, this leaves a little more room for growth.

Reply:
Would you help to give one possible scenario?

Reply:
I thought about this all day and I don't have a great scenario for you. I thought setting a propagator on TracerProvider might be useful. In that case, instrumentation would need something like this. For MeterProvider maybe exporter will need to get at labels? If we ever wanted to do something like have options on the provider that could be cross-cutting? It's basically enabling lower-level things to access the parent. Might end up being useful, might only ever be used for resources.